### PR TITLE
handle file length based on current content

### PR DESCRIPTION
**Pull request by srdjastankovic**

### CODE REVIEW

File: sources/json_parser.c [hunk 1]
```diff
@@ -524,8 +524,13 @@ bool json_deserialize_file_management_parameter(char* buffer, size_t buffer_size
                 return false;
             }
 
-            file_management_parameter_set_filename(parameter, value_buffer);
-            i++;
+            if (strlen(value_buffer) > FILE_MANAGEMENT_FILE_NAME_SIZE) {
+                /* Leave file name array empty */
+                return true;
```
Comment (milanzlicic) [inline sources/json_parser.c:529]: Why true?

Comment (srdjastankovic) [reply sources/json_parser.c:529]: parser vise it is legal state, nothing breaks, and I based check on the array length so I need true to start handling:
```
  const size_t num_deserialized_parameter = parser_deserialize_file_management_parameter(
      &ctx->parser, (char*)payload, (size_t)payload_len, &file_management_parameter);
  if (num_deserialized_parameter != 0) {
      _handle_file_management_parameter(&ctx->file_management, &file_management_parameter);
```
NOTE: parser that you looking at is called by parser_deserialize_file_management_parameter

File: sources/wolk_connector.c [hunk 2]
```diff
@@ -530,7 +530,9 @@ WOLK_ERR_T wolk_add_numeric_sensor_reading(wolk_ctx_t* ctx, const char* referenc
 
     char value_str[READING_SIZE];
     memset(value_str, 0, sizeof(value_str));
-    sprintf(value_str, "%f", value);
+    if (snprintf(value_str, READING_SIZE, "%f", value) > READING_SIZE) {
```
Comment (LazarBozic94) [inline sources/wolk_connector.c:533]: This condition checks if data is truncated. Should it be this way?

File: sources/wolk_connector.c [hunk 3]
```diff
@@ -561,7 +563,9 @@ WOLK_ERR_T wolk_add_multi_value_numeric_sensor_reading(wolk_ctx_t* ctx, const ch
     for (uint32_t i = 0; i < values_size; ++i) {
         char value_str[READING_SIZE];
         memset(value_str, 0, sizeof(value_str));
-        sprintf(value_str, "%f", values[i]);
+        if (snprintf(value_str, READING_SIZE, "%f", values[i]) > READING_SIZE) {
```
Comment (LazarBozic94) [inline sources/wolk_connector.c:566]: Same applies

**Review by LazarBozic94 — APPROVED** [review]

**Pull request merged**